src/unicode.cpp

```diff
@@ -27,4 +27,15 @@ namespace ada::unicode {
     return is_ascii_digit(c) || is_ascii_alpha(c);
   }
 
+  // A C0 control or space is a C0 control or U+0020 SPACE.
+  // A C0 control is a code point in the range U+0000 NULL to U+001F INFORMATION SEPARATOR ONE, inclusive.
+  bool is_c0_control_or_space(const char c) {
+    return c >= '\0' && c <= ' ';
```

Comment:
This is fine, but note that it is equivalent to...
```
return (unsigned char) c <= ' ';
```

(Note that the compiler should know this as well.)